enhancements/api-review/add-ImagePolicy-to-config.openshift.io-v1alpha1.md

```diff
@@ -0,0 +1,336 @@
+replaces:
+  - "/enhancements/that-less-than-great-idea.md"
+superseded-by:
+  - "/enhancements/our-past-effort.md"
+---
+
+# Add CRD ImagePolicy to config.openshift.io/v1alpha1
+
+## Summary
+
+This enhancement propose an ImagePolicy CRD to provide a way to set image signature verification configuration on Openshift.
+
+## Motivation
+
+### User Stories
+
+As an OpenShift user, I want to verify the container images signed using sigstore
+tools, so that I can utilize the increased security of my software supply chain.
+
+The user can create an ImagePolicy instance specifying the images/repositories to be verified and their policy. The MCO will write the configuration for signature verification. Once this is done, CRI-O can verify the images/repositories.
+
+### Goals
+
+- MCO container runtime config controller watches ImagePolicy instance in different namespaces and merges the instances for each namespace into a single containers-policy.json in a predefined /path/to/policies/\<NAMESPACE\>.json.
+
+- CRI-O can verify Cosign signature signed images using configuration from ImagePolicy
+
+### Non-Goals
+
+ImagePolicy CRD does not implements cosign with custom Fulcio and Rekor instance.
+
+## Proposal
+
+### Workflow Description
+
+**cluster administrator** is a human user responsible for adding signature verification configurations in
+cluster scope.
+1. The cluster administrator creates ImagePolicy yaml file with `namespace: opesnhift-config`.
+2. The cluster administrator creates ImagePolicy instance using `oc create -f imagepolicy.yaml`.
+3. The cluster administrator can get the merged cluster scope policies from `status` of ImagePolicy instances. 
+4. The cluster administrator can delete the signature verification configuration by deleting its ImagePolicy instances. 
+
+**application administrator** is a human user responsible for
+adding signature verification configurations for specific namespace.
+
+1. The application administrator creates ImagePolicy yaml file for namespace that is not `namespace: opesnhift-config`
+2. The application administrator creates ImagePolicy instance using `oc create -f imagepolicy.yaml`.
+3. The application administrator can get the merged cluster policy and policy for certain namespace from `status` of ImagePolicy instances. 
+4. The application administrator can delete the signature verification configuration by deleting its ImagePolicy instances. 
+
+### API Extensions
+
+#### Type definitions
+
+```go
+type ImagePolicySpec struct {
+	// images holds images/repositories to be verified
+	Images []Image `json:"images"`
+	// policy defines the verification policy
+	Policy Policy `json:"policy"`
+}
+
+// Image defines the list of images assigned to a policy.
+// For more information about the format, see the document
+// about the location field:
+// https://github.com/containers/image/blob/main/
+// docs/containers-policy.json.5.md#docker
+type Image struct {
+	Hostname string `json:"hostname,omitempty"`
```

Comment:
I agree scope matches the doc.
BTW I changed this from `type Image string` because the `hostname` from [googledoc](https://docs.google.com/document/d/16MQg6yVmvp1xkkgKAa1fUyCAbxZeq916IXznLAJmg6Q/edit?disco=AAAAxC2lpEc).
I understand it as this field will not include a tag or digest, since the cosign-created signatures only contain a repository?

we can just define this `type Image string`, without the key `scope`, and define the format of this field by using the kubebuilder regex and documents. Based on the above [comment](https://github.com/openshift/enhancements/pull/1402#discussion_r1196061110), we prefer without adding the `scope`?